src/main/java/uk/gov/hmcts/darts/transcriptions/auditing/TranscriptionUpdateAuditActivityProvider.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.darts.transcriptions.auditing;
+
+import uk.gov.hmcts.darts.audit.api.AuditActivity;
+import uk.gov.hmcts.darts.audit.api.AuditActivityProvider;
+import uk.gov.hmcts.darts.common.entity.TranscriptionEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionWorkflowEntity;
+import uk.gov.hmcts.darts.transcriptions.model.UpdateTranscriptionRequest;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.Comparator.comparing;
+import static uk.gov.hmcts.darts.audit.api.AuditActivity.AMEND_TRANSCRIPTION_WORKFLOW;
+
+public class TranscriptionUpdateAuditActivityProvider implements AuditActivityProvider {
+
+    public static TranscriptionUpdateAuditActivityProvider auditActivitiesFor(TranscriptionEntity entity, UpdateTranscriptionRequest patch) {
+        return new TranscriptionUpdateAuditActivityProvider(entity, patch);
+    }
+
+    private final Set<AuditActivity> auditActivities = new HashSet<>();
+
+    private TranscriptionUpdateAuditActivityProvider() {
+    }
+
+    private TranscriptionUpdateAuditActivityProvider(TranscriptionEntity entity, UpdateTranscriptionRequest patch) {
+        if (isWorkflowStatusTransitioning(entity, patch)) {
+            auditActivities.add(AMEND_TRANSCRIPTION_WORKFLOW);
+        }
+    }
+
+    @Override
+    public Set<AuditActivity> getAuditActivities() {
+        return auditActivities;
+    }
+
+    private boolean isWorkflowStatusTransitioning(TranscriptionEntity entity, UpdateTranscriptionRequest patch) {
+        var latestWorkflow = entity.getTranscriptionWorkflowEntities().stream()
+            .min(comparing(TranscriptionWorkflowEntity::getWorkflowTimestamp));
+
+        return latestWorkflow.isEmpty()
+            || !Objects.equals(latestWorkflow.get().getTranscriptionStatus().getId(), patch.getTranscriptionStatusId());
+    }
```

Comment:
this method could be on the `TranscriptionEntity` itself, if we are ok to have rich entities that are not just dump data holders

Reply:
It could, but so could many other methods in the codebase and we generally haven't taken that approach on this project.  I prefer a rich model design tbh especially instead of the endless private service method approach. Happy to discuss further.

Reply:
If I had to chose between having this method on the entity and code duplication I'd go for having it in the entity.
I've seen a lot of code duplication in our codebase. Code duplication is hell